libethereum/EthereumPeer.cpp

```diff
@@ -88,8 +86,7 @@ void EthereumPeer::setIdle()
 
 void EthereumPeer::requestStatus()
 {
-	if (m_asking != Asking::Nothing)
-	clog(NetWarn) << "Bad state: requesting state should be the first action";
+	assert(m_asking == Asking::Nothing);
```

Comment:
This is programming invariant, not a runtime error we can recover from. requestStatus is private and called from constructor only. If assert fails here this is bad programming, and better Fail Fast